Nop.Plugin.Shipping.USPS/Services/USPSService.cs

```diff
@@ -492,14 +492,23 @@ private bool IsDomesticRequest(GetShippingOptionRequest getShippingOptionRequest
                 return (shippingOptions, response.Packages.Select(x => $"Error Desc: {x.Error.Description}. USPS Help Context: {x.Error.HelpContext}."));
             }
 
-            shippingOptions.AddRange(response.Packages
-                .SelectMany(x => x.Postage
-                    .Where(isPostageOffered)
-                    .Select(p => new ShippingOption
+            List<int> shippingIdsAdded = new List<int>();
+            foreach (USPSPackageBase responsePackage in response.Packages)
```

Comment:
Changes from linq to foreach because the single linq statment was getting far too complex.

Reply:
I didn’t test it with like the bug description. Unsure if USPS api uses the same ID for each rate. I was personally seeing 10+ identical items for shipping internationally, all with ID “2”.